### PR TITLE
fix: Add 'base.' to make sure we not access generated property with same base name

### LINKED ISSUE #5054: Child control with x:Name="Content" causes rendering issue or compile fail
## Current behavior

I am seeing an "interesting" edge case on with the XAML Code generator... If I have a child control within a panel and give it an `x:Name="Content"` one of two things happen - if the child control is the same type as the parent panel, everything compiles, but nothing renders on run. If, however, the top level panel is a `Grid` and the child control is a `TextBlock,` then the following compile error occurs:

```
Severity    Code    Description    Project    File    Line    Suppression State
Error    CS0029    Cannot implicitly convert type 'Windows.UI.Xaml.Controls.Grid' to 'Windows.UI.Xaml.Controls.TextBlock'    TimeEntryUno.Wasm    D:\D-Repos\Uno.Silverlight\TimeEntryUno\TimeEntryUno.Wasm\obj\Debug\netstandard2.0\g\XamlCodeGenerator\TimeEntryPage_1ebe7e054bc16c7ccec314a1f21e1952.g.cs    57    Active
```

Issue is caused  because there's a "Content" property on the Page control and the "Content" local field of type TextBlock, is hiding the property from the base class.

image

This property in the generated code is the issue:

image

## Expected behavior

The code compiles and/or renders for the other heads as it does for UWP.

## How to reproduce it (as minimally and precisely as possible)

Fails to compile:
```xml
    <Grid>
        <TextBlock Text="Placeholder stuff" x:Name="Content"/>
    </Grid>
```

Compiles but fails to render:
```xml
    <Grid>
        <Grid x:Name="Content">
            <TextBlock Text="Placeholder stuff" x:Name="MessageText"/>
        </Grid>
    </Grid>
```

## Workaround

Use a different `x:Name` value!

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #5054

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

When the source generator is generating a property with name "Content", it shadows the base `Content` property, which causes issues when it's meant to access the base "Content" property.

## What is the new behavior?

Access the property with `base.` prefix to make sure we access the desired property.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/XamlGenerationTests/XamlGenerationTests.csproj [hunk 1]
```diff
@@ -22,22 +22,23 @@
 	</PropertyGroup>
 
 	<ItemGroup Condition=" '$(TargetFramework)' == 'MonoAndroid11.0'">
-		<PackageReference Include="Xamarin.AndroidX.Legacy.Support.v4"/>
-		<PackageReference Include="Xamarin.AndroidX.AppCompat"/>
-		<PackageReference Include="Xamarin.AndroidX.RecyclerView"/>
+		<PackageReference Include="Xamarin.AndroidX.Legacy.Support.v4" />
+		<PackageReference Include="Xamarin.AndroidX.AppCompat" />
+		<PackageReference Include="Xamarin.AndroidX.RecyclerView" />
```
Comment (Youssef1313) [inline src/SourceGenerators/XamlGenerationTests/XamlGenerationTests.csproj:27]: Note: These are not manual changes. Visual Studio has automatically made them.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 2]
```diff
@@ -2204,7 +2204,7 @@ private bool BuildProperties(IIndentedStringBuilder writer, XamlObjectDefinition
 											if (implicitContentChild.Objects.Count == 1
 												&& SymbolEqualityComparer.Default.Equals(contentProperty.Type, FindType(implicitContentChild.Objects[0].Type)))
 											{
-												writer.AppendFormatInvariant(contentProperty.Name + " = ");
+												writer.AppendFormatInvariant($"base.{contentProperty.Name} = ");
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:2207]:
```suggestion
												writer.AppendFormatInvariant(contentProperty.Name + " = ");
```

**Comment by jeromelaban** [comment]
Looks like there are still some CI failures.

**Comment by Youssef1313** [comment]
@jeromelaban Do the failing CI checks correspond to a specific solution filter? I'm currently on `Uno.UI-Android-only.slnf` and the solution builds without errors.

**Comment by jeromelaban** [comment]
It's in the build for `src\SourceGenerators\XamlGenerationTests`. You'll need to add those locally to your filter to test it, they're not part of any filter.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 3]
```diff
@@ -5326,8 +5331,8 @@ private void TrySetParsing(IIndentedStringBuilder writer, XamlObjectDefinition o
 			TryAnnotateWithGeneratorSource(writer);
 			if (HasIsParsing(objectDefinition.Type))
 			{
-				writer.AppendLineInvariant("IsParsing = true");
-				writer.AppendLineInvariant(isInitializer ? "," : ";");
+				// Append [base.]IsParsing = true(, or ;)
+				writer.AppendLineInvariant($"{(!isInitializer ? "base." : string.Empty)}IsParsing = true{(isInitializer ? "," : ";")}");
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:5335]: Let's decompose this for readability:

```suggestion
                var basePrefix = !isInitializer ? "base." : string.Empty;
                var initializerSuffix = isInitializer ? "," : ";";
				writer.AppendLineInvariant($"{basePrefix }IsParsing = true{initializerSuffix}");
```

**Review by jeromelaban — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**